walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> overlapEvents = new ArrayList<Event>();
+    List<TimeRange> overlapRanges = new ArrayList<TimeRange>();
+    List<TimeRange> noOverlapRanges = new ArrayList<TimeRange>();
+    List<TimeRange> possibleRanges = new ArrayList<TimeRange>();
+    int requestDuration = (int) request.getDuration();
+
+    if (requestDuration > TimeRange.WHOLE_DAY.duration()) {
+      return possibleRanges;
```

Comment:
Since this check says "if the request is impossible return no possibilities" e.g. an edge case, I would actually return a literal empty list instead of a variable that happens to be empty. (If this were the base case to an inductive formula, then I could agree that returning the variable makes sense.)